tools/cli/reserved-name.ts

```diff
@@ -0,0 +1,81 @@
+import { JWKInterface } from 'arweave/node/lib/wallet';
+import inquirer from 'inquirer';
+
+import { IOState } from '../../src/types';
+import {
+  arnsContractTxId,
+  getContractManifest,
+  initialize,
+  loadWallet,
+  warp,
+} from '../utilities';
+import questions from './questions';
+
+(async () => {
+  // simple setup script
+  initialize();
+
+  // Get the key file used for the distribution
+  const wallet: JWKInterface = loadWallet();
+
+  const reservedName = await inquirer.prompt(questions.createReservedName());
+
+  // get contract manifest
+  const { evaluationOptions = {} } = await getContractManifest({
+    contractTxId: arnsContractTxId,
+  });
+
+  // Connect the ArNS Registry Contract
+  const contract = await warp
+    .contract<IOState>(arnsContractTxId)
+    .connect(wallet)
+    .setEvaluationOptions(evaluationOptions)
+    .syncState(`https://api.arns.app/v1/contract/${arnsContractTxId}`, {
+      validity: true,
+    });
+
+  const confirm = await inquirer.prompt({
+    name: 'confirm',
+    type: 'confirm',
+    message: `CONFIRM RESERVED NAME DETAILS? ${JSON.stringify(reservedName)} >`,
+  });
+
+  if (confirm.confirm) {
+    const payload = {
+      function: 'createReservedName',
+      target: reservedName.target,
+      name: reservedName.name,
+      endTimestamp: reservedName.endTimestamp,
+    };
+
+    let attempts = 0;
+    let lastErrorMessage;
+    while (attempts < 3) {
+      const dryWrite = await contract.dryWrite(payload);
+      if (dryWrite.type === 'error' || dryWrite.errorMessage) {
+        lastErrorMessage = dryWrite.errorMessage;
```

Comment:
can this be null?
```suggestion
      if (dryWrite.type === 'error' || dryWrite.errorMessage) {
        lastErrorMessage = dryWrite.errorMessage ?? "Unknown error";
```

Reply:
[19b5c2f](https://github.com/ar-io/testnet-contract/pull/240/commits/19b5c2fa38ff008278d96cc2ee56e00bd7d8be76)